homeassistant/components/samsungtv/config_flow.py

```diff
@@ -78,15 +77,76 @@ def _get_entry(self):
             data=data,
         )
 
+    async def _async_set_device_unique_id(self, raise_on_progress=True):
+        """Set device unique_id."""
+
+        if self._id:
+            await self.async_set_unique_id(
+                self._id, raise_on_progress=raise_on_progress
+            )
+            self._abort_if_unique_id_configured()
```

Comment:
We should update the ip address if its changed.  We will need to pass it in.

```suggestion
            self._abort_if_unique_id_configured(updates={CONF_IP_ADDRESS: ip_address})
```

Reply:
I blindly accepted, but nothing refer anymore to ip address.
Why you think this update is need ?

Simone

Reply:
It should update CONF_HOST instead

Reply:
`self._abort_if_unique_id_configured(updates={CONF_HOST: self._host})`. We can actually do it everywhere since we are validating it